src/rgw/rgw_data_sync.cc

```diff
@@ -2659,8 +2743,13 @@ int RGWRunBucketSyncCoroutine::operate()
           << " failed, retcode=" << retcode << dendl;
       return set_cr_error(retcode);
     }
+
     yield {
       if ((rgw_bucket_shard_sync_info::SyncState)sync_status.state == rgw_bucket_shard_sync_info::StateFullSync) {
+	if (!bucket_info.datasync_flag_enabled()) { 
+	  ldout(sync_env->cct, 5) << "bucket sync stoped on" << bucket_str{bs.bucket} << dendl;
+	  return set_cr_done();
+	}
```

Comment:
i think we still need to remove the sync status object in this case. when we re-enable sync on this bucket, we'd want to restart full sync from the beginning instead of the current `rgw_bucket_shard_full_sync_marker::position`

you can have `RGWBucketShardFullSyncCR` check this directly after the `set_status("lock acquired");`